payment/datastore.go

```diff
@@ -403,6 +406,77 @@ func (pg *Postgres) IsStripeSub(orderID uuid.UUID) (bool, string, error) {
 	return ok, "", err
 }
 
+// RenewOrder updates the orders status to paid and paid at time, inserts record of this order
+// 	Status should either be one of pending, paid, fulfilled, or canceled.
+func (pg *Postgres) RenewOrder(ctx context.Context, orderID uuid.UUID) error {
+	// create tx
+	tx, err := pg.RawDB().BeginTxx(ctx, nil)
+	if err != nil {
+		return err
+	}
+	defer pg.RollbackTx(tx)
+
+	// how long should the order be valid for?
+	var orderTimeBounds = struct {
+		ValidFor *time.Duration `db:"valid_for"`
+		LastPaid sql.NullTime   `db:"last_paid"`
+	}{}
+
+	err = tx.GetContext(ctx, &orderTimeBounds, `
+		SELECT valid_for, last_paid
+		FROM orders
+		WHERE id = $1
+	`, orderID)
+	if err != nil {
+		return fmt.Errorf("unable to get order time bounds: %w", err)
+	}
+
+	// update last paid
+	lastPaid := time.Now()
+	var expiresAt time.Time
+	if orderTimeBounds.ValidFor != nil && orderTimeBounds.LastPaid.Valid {
+		expiresAt = orderTimeBounds.LastPaid.Time.Add(*orderTimeBounds.ValidFor)
```

Comment:
should this be `lastPaid.Add()` instead? it seems like this would be setting expiresAt based on the previous payment timestamp

Reply:
:facepalm: corrected: https://github.com/brave-intl/bat-go/pull/894/commits/c600e68352dea35c44e26d10084bd27dedff2cbf